javascript.md

```diff
@@ -0,0 +1,793 @@
+1. [Object Literals](#object-literals)
+1. [Array Literals](#array-literals)
+1. [Regular Expressions](#regular-expressions)
+1. [`console` Statements](#console-statements)
+1. [Comments](#comments)
+1. [Variable Naming](#variable-naming)
+1. [Polyfills](#polyfills)
+1. [Everyday Tricks](#everyday-tricks)
+1. [Credits](#credits)
+
+## Linting
+
+Lint your code using ESLint. [CFPB-tailored ESLint options](https://github.com/cfpb/front-end/js-standards/.eslintrc) exist.
+
+## Modules
+
+This style guide assumes you're using a module system such as [CommonJS][1], [AMD][2], [ES6 Modules][3], or any other kind of module system. Modules systems provide individual scoping, avoid leaks to the `global` object, and improve code base organization by **automating dependency graph generation**, instead of having to resort to manually creating tens of `<script>` tags.
+
+Module systems also provide us with dependency injection patterns, which are crucial when it comes to testing individual components in isolation.
+
+## Strict Mode
+
+**Always** put [`'use strict';`][4] at the top of your modules. Strict mode allows you to catch nonsensical behavior, discourages poor practices, and _is faster_ because it allows compilers to make certain assumptions about your code.
+
+## Spacing
+
+Spacing must be consistent across every file in the application. To this end, using something like [`.editorconfig`][5] configuration files is highly encouraged. Here are the defaults we suggest to get started with JavaScript indentation.
+
+```ini
+# editorconfig.org
+root = true
+
+[*]
+indent_style = space
+indent_size = 2
+end_of_line = lf
+charset = utf-8
+trim_trailing_whitespace = true
+insert_final_newline = true
+
+[*.md]
+trim_trailing_whitespace = false
+```
+
+We recommend using 2 spaces for indentation. The `.editorconfig` file can take care of that for us and everyone would be able to create the correct tabs by pressing the tab key.
+
+Spacing doesn't just entail tabbing, but also the spaces before, after, and in between arguments of a function declaration. This kind of spacing is typically highly irrelevant to get right, and it'll be hard for most teams to even arrive at a scheme that will satisfy everyone.
+
+```js
+function () {}
+```
+
+```js
+function( a, b ){}
+```
+
+```js
+function(a, b) {}
+```
+
+```js
+function (a,b) {}
+```
+
+Try to keep these differences to a minimum, but don't put much thought to it either.
+
+Where possible, improve readability by keeping lines below the 80-character mark.
+
+## Naming
+
+- Functions, variables, methods, objects and instances should be named using `camelCase`.
+- Constructors and prototypes should use `UpperCamelCase`.
+- Symbolic constants should use `UPPERCASE`.
+
+## Semicolons`;`
+
+The majority of JavaScript developers [prefer using semicolons][6] at the end of lines and we recommend following this practice. This choice is done to avoid potential issues with Automatic Semicolon Insertion _(ASI)_.
```

Comment:
Perhaps leave off `The majority of JavaScript developers` and just say we advocate using a semicolon. `The majority` seems wishy-washy without anything to back it up, and could change.

Personally I use semicolons, but I did see a pattern that used a leading semicolon to avoid issues with ASI:

```
;(function(){
...
})();
```